src/EditorFeatures/Core/DocumentationComments/DocumentationCommentSuggestion.cs

```diff
@@ -0,0 +1,77 @@
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ApplicationInsights.DataContracts;
+using Microsoft.VisualStudio.Language.Proposals;
+using Microsoft.VisualStudio.Language.Suggestions;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.DocumentationComments
+{
+    internal class DocumentationCommentSuggestion(AbstractDocumentationCommentCommandHandler handlerInstance, ProposalBase proposal) : SuggestionBase
+    {
+        public ProposalBase Proposal { get; } = proposal;
+
+        public override TipStyle TipStyle => TipStyle.AlwaysShowTip;
+
+        public override EditDisplayStyle EditStyle => EditDisplayStyle.GrayText;
+
+        public override bool HasMultipleSuggestions => false;
+
+        public override event PropertyChangedEventHandler? PropertyChanged;
+
+        public override async Task OnAcceptedAsync(SuggestionSessionBase session, ProposalBase originalProposal, ProposalBase currentProposal, ReasonForAccept reason, CancellationToken cancel)
+        {
+            var threadingContext = handlerInstance.ThreadingContext;
+            if (threadingContext != null)
+            {
+                await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancel);
+                await handlerInstance.DisposeAsync().ConfigureAwait(false);
+            }
+        }
+
+        public override Task OnChangeProposalAsync(SuggestionSessionBase session, ProposalBase originalProposal, ProposalBase currentProposal, bool forward, CancellationToken cancel)
+        {
+            return Task.CompletedTask;
+        }
+
+        public override async Task OnDismissedAsync(SuggestionSessionBase session, ProposalBase? originalProposal, ProposalBase? currentProposal, ReasonForDismiss reason, CancellationToken cancel)
+        {
+            var threadingContext = handlerInstance.ThreadingContext;
+            if (threadingContext != null)
+            {
+                await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancel);
+                await handlerInstance.ClearSuggestionAsync(reason, cancel).ConfigureAwait(false);
+            }
+        }
+
+        public override Task OnProposalUpdatedAsync(SuggestionSessionBase session, ProposalBase? originalProposal, ProposalBase? currentProposal, ReasonForUpdate reason, VirtualSnapshotPoint caret, CompletionState? completionState, CancellationToken cancel)
+        {
+            if (reason.HasFlag(ReasonForUpdate.Diverged))
+            {
+                return session.DismissAsync(ReasonForDismiss.DismissedAfterBufferChange, cancel);
+            }
+
+            return Task.CompletedTask;
+        }
+    }
+
+    internal class DocumentationCommentHandlerProposal(VirtualSnapshotPoint snapshotPoint, IReadOnlyList<ProposedEdit> edits) : ProposalBase
```

Comment:
Saw this [Proposal](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=/src/Editor/Language/Def/Language/Proposals/Proposal.cs&version=GBmain&_a=contents&line=19&lineStyle=plain&lineEnd=20&lineStartColumn=1&lineEndColumn=1) type in platform, can we use it instead?

Reply:
updated